### DISCUSSION TITLE
Should I use uv_timer_t for each connections?

### DISCUSSION THREAD

**Discussion by vfssoft** [libuv/libuv#4528] [General]
I'm developing the TCP server based on libuv, and I'm trying to figure out what's the correct way to implement an idle timeout mechanism for my server. 

The first thought that came to my mind was:

Use a `uv_timer_t` for each connected TCP connection, and start the timer with the connection **idle timeout**. Update the timer each time when the socket tickles.

My question is: Will there be any performance issues with this implementation? `uv_timer_t`  is heavy or lightweight to use in my case?

Thanks for any help!

**Comment by bnoordhuis** [comment]
That's the standard solution and it's lightweight; `uv_timer_t` is implemented in terms of a minheap.

**Reply by vfssoft** [reply]
Thanks for your reply, it helps a lot. 

One more question: There is no way to update the timeout value after starting the timer. So the recommended way is to stop the current timer and then start it again with the new timeout value?

**Reply by bnoordhuis** [reply]
Yes, stop and start is how you do that.
